assets/src/block-editor/components/amp-preview.js

```diff
@@ -0,0 +1,318 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createRef, renderToString } from '@wordpress/element';
+import { Icon, IconButton } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { DotTip } from '@wordpress/nux';
+import { compose } from '@wordpress/compose';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import ampBlackIcon from '../../../images/amp-black-icon.svg';
+import ampFilledIcon from '../../../images/amp-icon.svg';
+import { isAMPEnabled } from '../helpers';
+import { POST_PREVIEW_CLASS } from '../constants';
+
+/**
+ * Writes the message and graphic in the new preview window that was opened.
+ *
+ * Forked from the Core component <PostPreviewButton>.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-preview-button/index.js#L17
+ * @param {Object} targetDocument The target document.
+ */
+function writeInterstitialMessage( targetDocument ) {
+	let markup = renderToString(
+		<div className="editor-post-preview-button__interstitial-message">
+			<Icon
+				icon={ ampBlackIcon( { viewBox: '0 0 98 98' } ) }
+			/>
+			<p>
+				{ __( 'Generating AMP preview…', 'amp' ) }
+			</p>
+		</div>
+	);
+
+	markup += `
+		<style>
+			body {
+				margin: 0;
+			}
+			.editor-post-preview-button__interstitial-message {
+				display: flex;
+				flex-direction: column;
+				align-items: center;
+				justify-content: center;
+				height: 100vh;
+				width: 100vw;
+			}
+			@-webkit-keyframes paint {
+				0% {
+					stroke-dashoffset: 0;
+				}
+			}
+			@-moz-keyframes paint {
+				0% {
+					stroke-dashoffset: 0;
+				}
+			}
+			@-o-keyframes paint {
+				0% {
+					stroke-dashoffset: 0;
+				}
+			}
+			@keyframes paint {
+				0% {
+					stroke-dashoffset: 0;
+				}
+			}
+			.editor-post-preview-button__interstitial-message svg {
+				width: 198px;
+				height: 198px;
+				stroke: #555d66;
+				stroke-width: 0.75;
+			}
+			.editor-post-preview-button__interstitial-message svg .outer,
+			.editor-post-preview-button__interstitial-message svg .inner {
+				stroke-dasharray: 280;
+				stroke-dashoffset: 280;
+				-webkit-animation: paint 1.5s ease infinite alternate;
+				-moz-animation: paint 1.5s ease infinite alternate;
+				-o-animation: paint 1.5s ease infinite alternate;
+				animation: paint 1.5s ease infinite alternate;
+			}
+			p {
+				text-align: center;
+				font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
+			}
+		</style>
+	`;
+
+	targetDocument.write( markup );
```

Comment:
In the `<PostPreviewButton>` in Gutenberg that this is forked from, there's a [filter for markup](https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-preview-button/index.js#L88):

```js
markup = applyFilters( 'editor.PostPreview.interstitialMarkup', markup );
```

I decided not to add one here, but feel free to comment if you'd like one.